pkg/serviceaccount/ensure.go

```diff
@@ -45,8 +44,8 @@ const (
 
 func ensure(ctx context.Context, kubeClient kubernetes.Interface, namespace string, sa *corev1.ServiceAccount) (bool, error) {
 	result, err := util.CreateOrUpdate(ctx, resource.ForServiceAccount(kubeClient, namespace), sa,
```

Comment:
```suggestion
	result, err := util.CreateOrUpdate[*corev1.ServiceAccount](ctx, resource.ForServiceAccount(kubeClient, namespace), sa,
```

Reply:
I rather like letting the compiler infer the type :wink:

Reply:
yeah but you've specified the type explicitly in other places so for consistency.... but it doesn't matter either way, that's why I didn't -1 😄

Reply:
Ah, I thought I’d dropped the type parameters everywhere they weren’t needed, but I missed some.